src/mpio.jl

```diff
@@ -1,6 +1,14 @@
 using .MPI
 import Libdl
 
+const depsfile = joinpath(@__DIR__, "..", "deps", "deps.jl")
```

Comment:
The check here requires we check `libhdf5`, but since we moved that to `API` submodule I had to re-add this here.

Reply:
The tests should've shown this as an error, so this requires some debugging.